api/v1beta1/common_types.go

```diff
@@ -48,3 +48,35 @@ type CinderDebug struct {
 	// Service enable debug
 	Service bool `json:"service,omitempty"`
 }
+
+// CinderCephBackend defines the Ceph client parameters
+type CinderCephBackend struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="CephClusterFSID"
+	// CephClusterFSID defines the fsid
+	CephClusterFSID string `json:"cephFsid,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="CephClusterMonHosts"
+	// CephClusterMons defines the commma separated mon list
+	CephClusterMonHosts string `json:"cephMons,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="CephClientKey"
+	// CephClientKey set the Ceph cluster key used by Cinder
+	CephClientKey string `json:"cephClientKey,omitempty"`
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="CephUser"
+	// CephUser set the Ceph cluster pool used by Cinder
+	CephUser string `json:"cephUser,omitempty"`
+	// +kubebuilder:validation:Optional
+	// CephPools - Map of chosen names to spec definitions for the Ceph cluster
+	// pools used by Cinder
+	CephPools map[string]CephPoolSpec `json:"cephPools,omitempty"`
+}
+
+// CephPoolSpec defines the Ceph pool Spec parameters
+type CephPoolSpec struct {
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default="volumes"
+	// CephPoolName defines the name of the pool
+	CephPoolName string `json:"name,omitempty"`
+}
```

Comment:
Is the idea of using a struct here to support future parameters beyond just a name?  Because I guess otherwise the `CephPools` field in `CinderCephBackend` could just be a `map[string]string`.

Reply:
Yes, that's the reason.
In future we can have some overrides in place that can influence the way the service is configured (e.g. tiering depending on the `crush rules` set on the ceph cluster).
It's a good idea having this kind of struct here to help working on small changes (easier to review) in the follow up patches.

Reply:
Ack, sounds good